### PR TITLE
feat(List): Add support for custom heading id

### PR DESCRIPTION
**Pull request by Barsnes**
Adds support for passing a custom heading id

### CODE REVIEW

File: packages/react/src/components/List/List.tsx [hunk 1]
```diff
@@ -54,7 +59,7 @@ export const List = ({
         <Heading
           size={headingSize}
           level={headingLevel}
-          id={headingId}
+          id={headingId ?? hId}
```
Comment (mimarz) [inline packages/react/src/components/List/List.tsx:62]: Suggest making a separate `const` for the since its used multiple places.

**Review by mimarz — CHANGES_REQUESTED** [review]
We can add this for now, but maybe refactor this so you can use it with your own `Heading`. Exposing tunnel props doesn't exactly scale well. 

If we move the `...rest` further down (keeping `style` & `className`) to the ul/li element. People can then combine their own `Heading` and `List` with `aria-labelledby`

**Review by mimarz — APPROVED** [review]
Forgot to change submit type 🦭

**Comment by Barsnes** [comment]
> We can add this for now, but maybe refactor this so you can use it with your own `Heading`. Exposing tunnel props doesn't exactly scale well.
> 
> If we move the `...rest` further down (keeping `style` & `className`) to the ul/li element. People can then combine their own `Heading` and `List` with `aria-labelledby`

Yes I agree, we should not have a heading prop. I proposed this when making the list, but we decided not do it, since it would have to be client rendered for the `aria-labelled` by to work.

**Comment by Barsnes** [comment]
> We can add this for now, but maybe refactor this so you can use it with your own `Heading`. Exposing tunnel props doesn't exactly scale well.
> 
> If we move the `...rest` further down (keeping `style` & `className`) to the ul/li element. People can then combine their own `Heading` and `List` with `aria-labelledby`

I have made a PR with a potential change to how we use the `List`: #1400
Read the description for more thoughts

**Pull request merged**